<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>721</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QGridLayout" name="main_view_layout">
      <item row="0" column="0">
       <widget class="MainGraphicsView" name="main_view"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <addaction name="menu_file"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="select_view_dock">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="select_view_dock_contents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="select_view_dock_layout">
       <item>
        <widget class="SideGraphicsView" name="select_view"/>
       </item>
       <item>
        <widget class="QPushButton" name="view_selected_btn">
         <property name="text">
          <string>View selected</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_0">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="generate_word_btn_layout">
         <item>
          <widget class="QPushButton" name="generate_acceptable_btn">
           <property name="text">
            <string>Generate acceptable</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="generate_unacceptable_btn">
           <property name="text">
            <string>Generate unacceptable</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLineEdit" name="test_word_le">
         <property name="placeholderText">
          <string>Word</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="test_word_btn_layout">
         <item>
          <widget class="QPushButton" name="test_reset_btn">
           <property name="text">
            <string>&lt;&lt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="test_back_btn">
           <property name="text">
            <string>&lt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="test_forward_btn">
           <property name="text">
            <string>&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="test_end_btn">
           <property name="text">
            <string>&gt;&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="Line" name="line_1">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="generate_regex_layout">
         <item>
          <widget class="QPushButton" name="generate_regex_btn">
           <property name="text">
            <string>Generate RegEx</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="generate_regex_le">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="view_dock_info">
         <property name="text">
          <string/>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="construction_dock">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="construction_dock_contents">
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <layout class="QVBoxLayout" name="construction_dock_layout">
       <item>
        <widget class="QGroupBox" name="construction_by_element_gb">
         <property name="title">
          <string>By element</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="2" column="0">
           <widget class="QLabel" name="construction_by_element_info">
            <property name="text">
             <string/>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QGridLayout" name="construction_by_element_layout">
            <item row="3" column="0">
             <widget class="QLineEdit" name="final_states_le">
              <property name="placeholderText">
               <string>Final states</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLineEdit" name="alphabet_le">
              <property name="placeholderText">
               <string>Alphabet</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLineEdit" name="initial_states_le">
              <property name="placeholderText">
               <string>Initial states</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLineEdit" name="states_le">
              <property name="placeholderText">
               <string>States</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" rowspan="3">
             <widget class="QListWidget" name="transition_list">
              <property name="selectionMode">
               <enum>QAbstractItemView::MultiSelection</enum>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <layout class="QHBoxLayout" name="transition_layout">
              <item>
               <widget class="QLineEdit" name="transition_le">
                <property name="placeholderText">
                 <string>Transition</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="add_transition_btn">
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="remove_transition_btn">
                <property name="text">
                 <string>Remove</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="construction_by_element_btn">
            <property name="text">
             <string>Construct</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="construction_by_regex_gb">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>By RegEx</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QHBoxLayout" name="construction_by_regex_layout">
            <item>
             <widget class="QLineEdit" name="regex_construction_le">
              <property name="placeholderText">
               <string>RegEx</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="construction_by_regex_btn">
              <property name="text">
               <string>Construct</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="construction_by_regex_info">
            <property name="text">
             <string/>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="construction_dock_spacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MainGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>main_graphics_view.hpp</header>
  </customwidget>
  <customwidget>
   <class>SideGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header location="global">side_graphics_view.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
